### PR TITLE
[Backend] Follow & block

### PR DESCRIPTION
**Pull request by supiket**
Looking forward to your constructive reviews on follow & block API implementation.

The implementation is an addition to `profiles` app.

In addition to [Follow & Block API Documentation](https://github.com/bounswe/2021SpringGroup2/wiki/Follow-&-Block-API-Documentation), a `get users that blocked you` method is also introduced. I thought it would be helpful for displaying a "You are blocked from seeing this user" message in the profile a user that blocked another user. 

Please raise any issue that you think should be changed or fixed.

### CODE REVIEW

**Comment by supiket** [comment]
Thank you for adding makemigrations @ege-kaya!

File: backend/authentication/models.py
Comment (ege-kaya) [inline backend/authentication/models.py:66]: Just curious because I have never used a ManyToManyField, what does the 'self' argument mean?

File: backend/profiles/models.py
Comment (ege-kaya) [inline backend/profiles/models.py:3]: We will not leave this import, because of what I will promptly write below

File: backend/profiles/models.py
Comment (ege-kaya) [inline backend/profiles/models.py:9]: As we have noticed today, using `default=datetime.now()` causes a bug where all instances of this field display the same fixed date, the date on which this model was first created on the database. We figured out today that we should be using `auto_now_add=True` instead of a default value.

File: backend/profiles/models.py
Comment (ege-kaya) [inline backend/profiles/models.py:15]: Same issue as my above comment

File: backend/profiles/views.py
Comment (ege-kaya) [inline backend/profiles/views.py:214]: Seems like we don't need the `follow_date=datetime.now()` argument here, since it will be automatically set by Django if we give the argument `auto_now_add=True` to the model.

File: backend/profiles/views.py
Comment (ege-kaya) [inline backend/profiles/views.py:261]: Same as above

File: backend/profiles/views.py
Comment (ege-kaya) [inline backend/profiles/views.py:14]: Looks like we won't need this import as well

File: backend/authentication/models.py
Comment (supiket) [reply backend/authentication/models.py:66]: Since the all those are also `User` models, and this code is in the definition of the `User` model, `self` argument is used to refer to itself.

For instance, if I were to add a `User` as `ForeignKey`, I would still use `self` in the following way: 
`models.ForeignKey('self', on_delete=CASCADE)`

But since we need lists here, we are using `ManyToManyField`s and no option for what happens on delete.

At first I tried using `ArrayField(models.ForeignKey('self', on_delete=CASCADE), default=empty_list)`, but apparently that is not how it works. For foreign keys, `ManyToManyField`s are used.

File: backend/profiles/models.py
Comment (supiket) [reply backend/profiles/models.py:9]: Oh, I also thought `datetime.now()` was problematic, good thing we have a solution. Thanks!

**Review by ege-kaya — CHANGES_REQUESTED** [review]
I have left my comments on your code, only some minor issues. I have only read the code and haven't manually tested it by running it on local, assuming that you have, however if you want me to I can also give that a try. If you tested it and it worked, I trust your judgement.

**Comment by supiket** [comment]
> I have left my comments on your code, only some minor issues. I have only read the code and haven't manually tested it by running it on local, assuming that you have, however if you want me to I can also give that a try. If you tested it and it worked, I trust your judgement.

I think we can go over in today's meeting. Thank you for your reviews :)

**Comment by supiket** [comment]
I think this PR is ready to be merged. Waiting for approval :)

**Review by ege-kaya — APPROVED** [review]
I have done some final additions and tested the entire code. I'm certain it works now, this can be merged.

**Pull request merged**
